x-pack/legacy/plugins/siem/public/containers/detection_engine/signals/api.ts

```diff
@@ -62,89 +58,65 @@ export const updateSignalStatus = async ({
   status,
   signal,
 }: UpdateSignalStatusProps): Promise<unknown> => {
-  const response = await fetch(`${chrome.getBasePath()}${DETECTION_ENGINE_SIGNALS_STATUS_URL}`, {
+  const response = await npStart.core.http.fetch(DETECTION_ENGINE_SIGNALS_STATUS_URL, {
     method: 'POST',
     credentials: 'same-origin',
-    headers: {
-      'content-type': 'application/json',
-      'kbn-xsrf': 'true',
-    },
     body: JSON.stringify({ status, ...query }),
+    asResponse: true,
     signal,
   });
 
-  await throwIfNotOk(response);
-  return response.json();
+  await throwIfNotOk(response.response!);
+  return response.body!;
 };
 
 /**
  * Fetch Signal Index
  *
  * @param signal AbortSignal for cancelling request
  */
-export const getSignalIndex = async ({ signal }: BasicSignals): Promise<SignalsIndex | null> => {
-  const response = await fetch(`${chrome.getBasePath()}${DETECTION_ENGINE_INDEX_URL}`, {
+export const getSignalIndex = async ({ signal }: BasicSignals): Promise<SignalsIndex> => {
+  const response = await npStart.core.http.fetch<SignalsIndex>(DETECTION_ENGINE_INDEX_URL, {
     method: 'GET',
     credentials: 'same-origin',
-    headers: {
-      'content-type': 'application/json',
-      'kbn-xsrf': 'true',
-    },
     signal,
+    asResponse: true,
   });
-  if (response.ok) {
-    const signalIndex = await response.json();
-    return signalIndex;
-  }
-  const error = await parseJsonFromBody(response);
-  if (error != null) {
-    throw new SignalIndexError(error);
```

Comment:
@XavierM `SignalIndexError` and `PostSignalError` are currently unused on this branch. Is it acceptable to just reuse `throwIfNotOk`, or do we need more here?

Reply:
After talking with @XavierM, we do in fact want to be more discriminate with throwing errors, here. I reimplemented this functionality in 91be0c2.